From cd2f3171eb9c07efa293e39e4194c6c7b774a08f Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 7 Jul 2020 17:17:26 +0800
Subject: [PATCH] soc: rockchip: power-domain: Add missing kfree in error path

If 'rockchip_pm_add_one_domain' fails, then 'qos_is_need_init[0]'
should be freed, otherwise, there is a memory leak.

Fixes: 2047231d247d ("soc: rockchip: power-domain: support qos init")
Change-Id: Id2ec5c4b7cbc53a5491f348eeba1fd526018875d
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/soc/rockchip/pm_domains.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/soc/rockchip/pm_domains.c b/drivers/soc/rockchip/pm_domains.c
index 9ab01581baae..3cd7b3e24801 100644
--- a/drivers/soc/rockchip/pm_domains.c
+++ b/drivers/soc/rockchip/pm_domains.c
@@ -880,6 +880,7 @@ static int rockchip_pm_add_one_domain(struct rockchip_pmu *pmu,
 	return 0;
 
 err_unprepare_clocks:
+	kfree(qos_is_need_init[0]);
 	clk_bulk_unprepare(pd->num_clks, pd->clks);
 err_put_clocks:
 	clk_bulk_put(pd->num_clks, pd->clks);
-- 
2.35.3

